capstone-backend/src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -86,6 +86,38 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     } catch (Exception e) {
       System.err.println("Error: " + e.getMessage());
     }
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    try {
+
+      String userID = request.getParameter("id");
+      List<Map<String, Object>> user = new ArrayList<Map<String, Object>>();
+
+      if (userID == null) {
+        System.err.println("Error:\t" + "Null Id");
+        response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+        return;
+      }
 
+      Firestore db = getFirestore();
+
+      Query query = db.collection("users").whereEqualTo("id", userID);
```

Comment:
If you're searching by ID, shouldn't you use the primary key search rather than a filter? Check this out: https://firebase.google.com/docs/firestore/query-data/get-data#get_a_document.

Reply:
Changed.